server/database.go

```diff
@@ -943,3 +950,138 @@ func validateDatabaseLabelList(labelList []*api.DatabaseLabel, labelKeyList []*a
 
 	return nil
 }
+
+func restoreDatabaseEdit(databaseEdit *api.DatabaseEdit) (string, error) {
+	stmtList := []string{}
+
+	for _, createTableContext := range databaseEdit.CreateTableList {
+		createTableStmt := transformCreateTableContext(createTableContext, databaseEdit.EngineType)
+		stmt, err := restoreAstNode(createTableStmt)
+		if err != nil {
+			return "", err
+		}
+
+		if stmt != "" {
+			stmt += ";"
+		}
+		stmtList = append(stmtList, stmt)
+	}
+
+	return strings.Join(stmtList, ""), nil
+}
+
+func transformCreateTableContext(createTableContext *api.CreateTableContext, engineType db.Type) *ast.CreateTableStmt {
+	tableName := &ast.TableName{
+		Name: model.NewCIStr(createTableContext.Name),
+	}
+	createTableStmt := &ast.CreateTableStmt{
+		Table: tableName,
+	}
+
+	createTableOptions := []*ast.TableOption{}
+	if createTableContext.Engine != "" {
+		createTableOptions = append(createTableOptions, &ast.TableOption{
+			Tp:       ast.TableOptionEngine,
+			StrValue: createTableContext.Engine,
+		})
+	}
+	if createTableContext.CharacterSet != "" {
+		createTableOptions = append(createTableOptions, &ast.TableOption{
+			Tp:       ast.TableOptionCharset,
+			StrValue: createTableContext.CharacterSet,
+		})
+	}
+	if createTableContext.Collation != "" {
+		createTableOptions = append(createTableOptions, &ast.TableOption{
+			Tp:       ast.TableOptionCollate,
+			StrValue: createTableContext.Collation,
+		})
+	}
+	if createTableContext.Comment != "" {
+		createTableOptions = append(createTableOptions, &ast.TableOption{
+			Tp:       ast.TableOptionComment,
+			StrValue: createTableContext.Comment,
+		})
+	}
+	createTableStmt.Options = createTableOptions
+
+	columnDefs := []*ast.ColumnDef{}
+	for _, addColumnContext := range createTableContext.AddColumnList {
+		columnDef := transformAddColumnContext(addColumnContext, engineType)
+		columnDefs = append(columnDefs, columnDef)
+	}
+	createTableStmt.Cols = columnDefs
+
+	return createTableStmt
+}
+
+func transformAddColumnContext(addColumnContext *api.AddColumnContext, engineType db.Type) *ast.ColumnDef {
+	colName := &ast.ColumnName{
+		Name: model.NewCIStr(addColumnContext.Name),
+	}
+	columnDef := &ast.ColumnDef{
+		Name: colName,
+	}
+
+	var colType *types.FieldType
+	if engineType == db.MySQL {
```

Comment:
MySQL and PG may have an entirely different implementations. Let’s assume all current usage applies to MySQL.